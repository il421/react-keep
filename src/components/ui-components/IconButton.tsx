import React from "react";
import {
  FontAwesomeIcon,
  FontAwesomeIconProps
} from "@fortawesome/react-fontawesome";
import "../../styles/ui-components/_icon-button.scss";

interface IconButtonProps extends FontAwesomeIconProps {
  onClick: (evt?: React.MouseEvent) => void;
  className?: string;
  disabled?: boolean;
}

export class IconButton extends React.PureComponent<IconButtonProps> {
  render() {
    const { onClick, className, disabled, ...rest } = this.props;
    return (
      <button
        className={`${className} icon-button`}
        onClick={onClick}
        disabled={disabled}
      >
        <FontAwesomeIcon {...rest} />
      </button>
    );
  }
}
