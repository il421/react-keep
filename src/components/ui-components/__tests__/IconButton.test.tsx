import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { IconButton, IconButtonProps } from "../IconButton";
import toJson from "enzyme-to-json";

let props: IconButtonProps, wrapper: ShallowWrapper;

beforeEach(() => {
  props = {
    icon: "times",
    text: "text",
  };
});

test("should render IconButton", () => {
  wrapper = shallow(<IconButton {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
