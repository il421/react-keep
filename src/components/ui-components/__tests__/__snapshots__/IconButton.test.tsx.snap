// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render IconButton 1`] = `
<button
  className="icon-button undefined "
  type="button"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="times"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
  <div
    className="icon-button__text"
  >
    text
  </div>
</button>
`;
